pkg/cloudprovider/provider/openstack/provider.go

```diff
@@ -27,32 +27,55 @@ import (
 )
 
 type provider struct {
-	privateKey *machinessh.PrivateKey
+	privateKey      *machinessh.PrivateKey
+	secretKeyGetter *providerconfig.SecretKeyGetter
+	rawConfig       RawConfig
```

Comment:
Don't save machine related state into the provider.
It does not belong into the provider

Reply:
We have to because of the defaulting in which parts of the `providerSpec` get updated. So far we just used the marshaled the `config` type for that, but we can not do that anymore because it contains the resolved values, thus we need to store the initial state somewhere.

Reply:
The provider is not the right place as it is not machine specific.
We rather should let `getConfig` return the resolved & the raw Config.

Reply:
Makes sense, I'll update it